test/lib/tools/get_integration_cards_guidelines/guidelines.ts

```diff
@@ -0,0 +1,46 @@
+import anyTest, {TestFn} from "ava";
+import sinonGlobal from "sinon";
+import esmock from "esmock";
+
+const fakeGuidelines = "# Guidelines\n\nSome guidelines content.";
+
+async function getMockedModule() {
+	const readFileStub = sinonGlobal.stub().resolves(fakeGuidelines);
+	const {getGuidelines} = await esmock("../../../../src/tools/get_integration_cards_guidelines/guidelines.ts", {
+		"node:fs/promises": {
+			readFile: readFileStub,
+		},
+	});
+	return {getGuidelines, readFileStub};
+}
+
+const test = anyTest as TestFn<{
+	sinon: sinonGlobal.SinonSandbox;
+	getGuidelines: typeof import("../../../../src/tools/get_integration_cards_guidelines/guidelines.js").getGuidelines;
+	readFileStub: sinonGlobal.SinonStub;
+}>;
+
+test.beforeEach(async (t) => {
+	t.context.sinon = sinonGlobal.createSandbox();
+	const {getGuidelines, readFileStub} = await getMockedModule();
+	t.context.getGuidelines = getGuidelines;
+	t.context.readFileStub = readFileStub;
+});
+
+test.afterEach.always((t) => {
+	t.context.sinon.restore();
+});
+
+test("getGuidelines returns guidelines content", async (t) => {
```

Comment:
may also directly end to end that the actual file is read and returns content. E.g. if file is renamed this test will continue to work, but the guidelines will not.
the test can assert for "any content" size>0 or for part of the content like the title.

or at least test if the file which was requested from readFileStub actually exists?

Reply:
I added new `guidelines.integration.ts` file, where I implemented such test